lib/utils.js

```diff
@@ -96,9 +96,9 @@ function parseParams(str) {
 
 function decodeText(text, textEncoding, destEncoding) {
   var ret;
-  if (text && jsencoding.encodingExists(destEncoding)) {
+  if (text && hasEncoding(destEncoding)) {
     try {
-      ret = jsencoding.TextDecoder(destEncoding)
+      ret = TextDecoder(destEncoding)
```

Comment:
would caching TextEncoders in a map here provide any performance benefits?

Reply:
It does not have a impact.